File: stuff_presentation/src/main/java/android/montreal/devfest/stuff/presentation/presenters/MainPresenter.java
package android.montreal.devfest.stuff.presentation.presenters;

public interface MainPresenter {
    void onViewReady();

    void updateRandomStuff(String text, int number);

    void openViewMoreLink();

    interface MainView {
        void showRandomStuff(String text, int number);

        void showUpdateSuccess();

        void showMissingError();

        void showUpdateError();
    }
}

File: stuff_presentation/src/main/java/android/montreal/devfest/stuff/presentation/presenters/MainPresenter.kt
package android.montreal.devfest.stuff.presentation.presenters

interface MainPresenter {
    fun onViewReady()

    fun updateRandomStuff(text: String, number: Int)

    interface MainView {
        fun showRandomStuff(text: String, number: Int)

        fun showUpdateSuccess()

        fun showMissingError()

        fun showUpdateError()
    }
}
